### DISCUSSION TITLE
Import query from .graphql file

### DISCUSSION THREAD

**Discussion by antonisntoulis** [0no-co/gql.tada#278] [Q&A] [answered]
Hello everyone, is it possible to import graphql queries from a .graphql file in a Vite project and be able to get types for the query result data?

```
import { graphql } from 'gql.tada';
import { useQuery } from 'urql';
import { get_users } from '@/graphql_queries/users.graphql';


const UsersQuery = graphql(get_users);

const [result] = useQuery({
  query: UsersQuery,
  variables: { limit: 5 },
});

console.log(result.data.users)

```

And be able to get type hints for the result data?

**Comment by kitten** [comment, marked as answer]
No, that's not possible.

The GraphQL documents must be embedded into TypeScript as string literals, and must be runtime readable as strings as well. The former is so the string type of a GraphQL document can be inferred to result and variables types, the latter is so that the document can be parsed into the AST format at runtime.

This page explains what Typed Documents in GraphQL are all about and goes through differences to codegen as well: https://gql-tada.0no.co/guides/typed-documents

Using `.graphql` documents, since they're separate from runtime code, always necessitates a different layer to take care of the transpilation and conversion, but in this case, we need and want the documents to be in TypeScript to get all the benefits of the:
- LSP
- type system & type inference
- and; co-location
